### ISSUE TITLE
[Bug] Repeatable fields no longer handles null values

### ISSUE THREAD

**Issue by jargoud** [Laravel-Backpack/CRUD#3649]
# Bug report

### What I did

I'm using a `repeatable` field in my form.

### What I expected to happen

When populating the subfield in the repeatable field, if the value is null, it should let the input empty.

### What happened

The `null` string value is populated.

### What I've already tried to fix it

I tried to add a ternary condition to firstly test the value before adding a prefix, [in repeatable field view, on line 227](https://github.com/Laravel-Backpack/CRUD/blob/master/src/resources/views/crud/fields/repeatable.blade.php#L227).

### Backpack, Laravel, PHP, DB version

> ### PHP VERSION:
> PHP 7.4.16 (cli) (built: Mar  4 2021 20:52:51) ( NTS )
> Copyright (c) The PHP Group
> Zend Engine v3.4.0, Copyright (c) Zend Technologies
>     with Zend OPcache v7.4.16, Copyright (c), by Zend Technologies
> 
> ### LARAVEL VERSION:
> v6.20.22@f4d2beff6f1793909e1db4f2204800183b4f1719
> 
> ### BACKPACK VERSION:
> 4.1.41@44770347b918e42c052d0a138c24535b43e32472
>

**Comment by pxpm** [comment]
Hello @jargoud 

Thanks for the report. 

Could you please share with me your `repeatable` field definition ? Or some POC code that I can use to reproduce ?

I tried with: 

```php
 $this->crud->addField([
            'name' => 'wysiwyg',
            'type' => 'repeatable',
            'fields' => [
                [
                    'name' => 'name',
                    'type' => 'text'
                ],
                [
                    'name' => 'something',
                    'type' => 'text'
                ]
            ]
        ]);
```

And I was not able to reproduce it, am I doing something wrong ? 

Best,
Pedro

**Comment by jargoud** [comment]
Hello @pxpm,

I don't see anything wrong with your field definition.

I'm not sure if that's relevant but I noticed the problem on an `url` typed subfield. On edition, when the value coming from the database was `null`, it was then casted to `"null"`.

As I said in my initial message, I fixed it in my project by using a ternary condition so that 

```
valuePrefix+values[$(this).data('repeatable-input-name')]
```

is done only if `values[$(this).data('repeatable-input-name')]` evaluates to `true`.

I'm not sure about how to provide a POC about this, do you have a guide?

Thanks,

Jeremy

**Comment by pxpm** [comment]
Hello @jargoud 

Thanks for clarification. I was able to reproduce your issue. You are probably overwritting the save process ? Because if you leave the field empty, it will just save an empty string in the database, not `null`. Anyway, that's not the point here 😄 

I'v taken your solution into consideration, but that would not be complete I think, if you have booleans in the table, when it's `false`, no prefix applied, if it's `true` prefix is applied. That could lead to other inconsistencies. 

In that regard I created PR #3654 that should solve your problem and account for some more scenarios. 

If you happen to give a try to that PR code, let us know in the comments, it could help to speed up the review and merging process. I am going to close this so we keep any discussion regarding this issue in the PR thread. 

Thanks again for your time and help,

Pedro
